### PR TITLE
feat: add REST API description

### LINKED ISSUE #14: Define the REST API
Define the public REST API exposed by `ratsd` to its clients.

The endpoints comprise, at least:

1. a client-authenticated challenge-response for obtaining the evidence
1. a `.well-known/ratsd` endpoint that provides information about the service (e.g., details about the topology of the composite attester - incl. evidence types -, service API catalogue, the status of the service, version of the installed software, etc.)

### PR DESCRIPTION
**Pull request by thomas-fossati**
Fix #14

### CODE REVIEW

File: docs/api/ratsd.yaml [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        - status
+      properties:
+        type:
+          type: string
+          enum:
+            - tag:github.com,2024:veraison/ratsd:error:unauthorized
+        title:
+          type: string
+          enum:
+            - access unautorized
```
Comment (thomas-fossati) [inline docs/api/ratsd.yaml:131]:
```suggestion
            - access unauthorized
```

File: docs/api/ratsd.yaml [hunk 2]
```diff
@@ -0,0 +1,148 @@
+openapi: 3.0.0
+info:
+  title: RATS Evidence Collection Daemon
+  version: 0.0.1
+tags: []
+paths:
+  /ratsd/chares:
+    post:
+      operationId: Ratsd_chares
```
Comment (deeglaze) [inline docs/api/ratsd.yaml:9]: What is chares other than the mythical ferryman? I’m trying to understand if 401 or 403 is the most appropriate error code

**Review by deeglaze — APPROVED** [review]

File: docs/api/ratsd.yaml (hunk 2 above)
Comment (thomas-fossati) [reply docs/api/ratsd.yaml:9]: At least for now, it seems sufficient to be successfully authenticated to request evidence using this challenge-response API.  I cannot imagine a case where users who have provided an authentication token that checks out aren't also allowed to query this interface.  In the future, we may add a 403 to handle such cases but for now 401 looks appropriate to me.

File: docs/api/ratsd.yaml [hunk 3]
```diff
@@ -0,0 +1,148 @@
+openapi: 3.0.0
+info:
+  title: RATS Evidence Collection Daemon
+  version: 0.0.1
+tags: []
+paths:
+  /ratsd/chares:
```
Comment (yogeshbdeshpande) [inline docs/api/ratsd.yaml:7]:
```suggestion
  /ratsd/chalrsp:
```

Comment (yogeshbdeshpande) [inline docs/api/ratsd.yaml:7]: I suggest we rename this to a more clear short version as chalrsp
which is more intuitive of what you intend to do.. Personally I struggled with what it means to be a chares?

File: docs/api/ratsd.yaml [hunk 4]
```diff
@@ -0,0 +1,148 @@
+openapi: 3.0.0
+info:
+  title: RATS Evidence Collection Daemon
```
Comment (yogeshbdeshpande) [inline docs/api/ratsd.yaml:3]:
```suggestion
  title: RATS Evidence Collection Daemon API
```

**Review by yogeshbdeshpande — COMMENTED** [review]
Some nits

File: docs/api/ratsd.yaml (hunk 3 above)
Comment (iolivergithub) [reply docs/api/ratsd.yaml:7]: Why not challengeReponse in full?  ( camelCase or whatever notwithstanding?)   

Is this interface only to be used for authentication tokens?

Comment (thomas-fossati) [reply docs/api/ratsd.yaml:7]: > Is this interface only to be used for authentication tokens?

What do you mean by "authentication tokens"?

Comment (thomas-fossati) [reply docs/api/ratsd.yaml:7]: > Personally I struggled with what it means to be a chares?


[Chares](https://en.wikipedia.org/wiki/Chares_of_Lindos) of Lindos, the sculptor of one of the seven wonders of the world!

File: docs/api/ratsd.yaml [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+  version: 0.0.1
+tags: []
+paths:
+  /ratsd/chares:
+    post:
```
Comment (jraman567) [inline docs/api/ratsd.yaml:8]:
```suggestion
    post:
      description: Challenge response API. Accepts a challenge and, upon success, generates the evidence in EAT w/ CMW format.
```

**Review by jraman567 — APPROVED** [review]
LGTM. Thank you, @thomas-fossati !

File: docs/api/ratsd.yaml (hunk 3 above)
Comment (iolivergithub) [reply docs/api/ratsd.yaml:7]: > > Is this interface only to be used for authentication tokens?
> 
> What do you mean by "authentication tokens"?

Just referring to an earlier statement.  What is expected in the POST body and what is returned?

Comment (thomas-fossati) [reply docs/api/ratsd.yaml:7]: An alternative is to use Greek in the URI (`/ratsd/Χάρης`) and the media type (`application/vnd.veraison.Χάρης+json`):

```http
POST /ratsd/Χάρης HTTP/1.1
Host: host.example
Accept: application/eat+jwt; eat_profile="tag:github.com,2024:veraison/ratsd"
Content-Type: application/vnd.veraison.Χάρης+json
Content-Length: 33

{
  "nonce": "MIDBNH28iioisjPy"
}
```

A fun conversation to have with IETF's Internationalization Directorate :-)

Comment (thomas-fossati) [reply docs/api/ratsd.yaml:7]: > What is expected in the POST body and what is returned?

The request body has an `application/vnd.veraison.chares+json` typed object:

```json
{ "nonce": "base64url_encoded_nonce_" }
```

The request will also sport the auth token in the `Authorization` header:

```http
Authorization: Bearer my.jwt.token
```

The response is a signed EAT with profile `tag:github.com,2024:veraison/ratsd`

**Pull request merged**